### PR TITLE
fix: memory leak in incremental assign

### PR DESCRIPTION
**Pull request by martijnimhoff**
The incremental assign contained a memory leak. This happened because when unassigning, the wrong new pointers are destroyed (m_partitions). 

Instead the `partitions` argument should be destroyed, because this is only used to find the right toppars in m_partitions. And the toppars in `m_partitions` which are unassigned also should be destroyed.

### CODE REVIEW

File: src/kafka-consumer.cc [hunk 1]
```diff
@@ -260,7 +263,8 @@ Baton KafkaConsumer::IncrementalUnassign(std::vector<RdKafka::TopicPartition*> p
   }
 
   // Destroy the old list of partitions since we are no longer using it
-  RdKafka::TopicPartition::destroy(m_partitions);
+  RdKafka::TopicPartition::destroy(delete_partitions);
```
Comment (milindl) [inline src/kafka-consumer.cc:266]: nit: update comment above this line to reflect these changes

**Review by milindl — COMMENTED** [review]
Thanks for the PR! Just a bunch of minor changes, looks good otherwise.

**Review by milindl — APPROVED** [review]
Approving and running CI

**Comment by martijnimhoff** [comment]
I didn't push properly, so the comments weren't added, but that is done now!

**Comment by milindl** [comment]
@martijnimhoff it looks good to me. As this is our first external PR with code changes, there's a bunch of minor things I'm doing to get the CI to run on this once, and there should be a message from our CLA bot in a while on this PR for you. Thanks!

**Comment by martijnimhoff** [comment]
I've signed it. Should be fine now?

**Comment by milindl** [comment]
Closing and reopening PR to try and trigger semaphore CI checks to run.

**Comment by milindl** [comment]
Thanks @martijnimhoff  for the fix!

**Pull request merged**